tests/integration/concurrent_clients/test_multithread_client.py

```diff
@@ -0,0 +1,55 @@
+
+import threading
+import time
+from concurrent.futures import ThreadPoolExecutor
+
+from jina import Client, Document, Executor, Flow, requests
+
+
+class MyExecutor(Executor):
+    @requests
+    def foo(self, docs, **kwargs):
+        for doc in docs:
+            doc.text = 'I am coming from MyExecutor'
+                
+def flow_run(flow, stop_event):
+    with flow:
+        flow.block(stop_event)
+
+
+def client_post(doc, client):
+    result = client.post(on='/', inputs=doc)[0]
+    return result
+
+
+def test_multithread_client(capsys):
+    stop_event = threading.Event()
+    flow = Flow(port=12345).add(uses=MyExecutor)
+    t = threading.Thread(target=flow_run, args=(flow, stop_event))
+
+    c = Client(port=12345)
+
+    try:
+        with capsys.disabled():
```

Comment:
what does this do?

Reply:
`capsys` will capture all the `stdout` and `stderr` from the test